torch/distributed/_tensor/api.py

```diff
@@ -105,16 +105,23 @@ def forward(  # type: ignore[override]
         device_mesh: DeviceMesh,
         placements: Tuple[Placement, ...],
         run_check: bool,
+        shape: Optional[torch.Size] = None,
+        stride: Optional[Tuple[int, ...]] = None,
     ) -> "DTensor":
         ctx.previous_placement = placements
         ctx.previous_device_mesh = device_mesh
 
-        # if it's not by default run_check, we assume user is certain that each
-        # rank has the same tensor shape, and we just use that to calculate the
-        # global shape
-        tensor_shape, tensor_stride = compute_global_tensor_info(
-            input, device_mesh, placements
-        )
+        if (shape is not None) != (stride is not None):
```

Comment:
hmmm this can be put to the below if condition? i.e. 
```
if shape and stride:
    tensor_shape, tensor_stride = shape, stride
elif shape is not None and stride is not None:
    ... compute global tensor info
else:
    raise RuntimeError

Reply:
Oh ye. Good catch! Definitely not thinking there. lol